confd/pkg/backends/calico/client.go

```diff
@@ -1342,7 +1342,19 @@ func (c *client) onClusterIPsUpdate(clusterCIDRs []string) {
 }
 
 func (c *client) onLoadBalancerIPsUpdate(lbIPs []string) {
-	if err := c.updateGlobalRoutes(c.loadBalancerIPs, lbIPs); err == nil {
+	var globalLbIPs []string
+	for _, lbIP := range lbIPs {
+		if strings.Contains(lbIP, ":") {
+			if !strings.HasSuffix(lbIP, "/128") {
```

Comment:
We'll also need a comment in this section explaining why we're doing this - it's not super obvious.

Something like this:

> We advertise the given LB IP ranges from every node in order to satisfy services with external traffic policy of "cluster". However, we don't want to advertise single IPs in this way because it breaks any "local" type services the user creates, which should instead be advertised from only a subset of nodes. So, we handle advertisement of any single-addresses found in the config on a per-service basis from within the routeGenerator.